### PR TITLE
Tested HeapVector containing HeapVector.

### PR DESCRIPTION
**Pull request by davidpiuva**
Completed the regression test of nested heap vectors.

### CODE REVIEW

File: test/collections/VectorTest.ooc [hunk 1]
```diff
@@ -214,12 +220,27 @@ VectorTest: class extends Fixture {
...
-				heapVector[i] = HeapVector<MyCover> new(sizeY)
-			heapVector free()
+			sizeX := 10
+			sizeY := 10
+			additionTable := HeapVector<HeapVector<MyClass>> new(sizeX)
+			for (i in 0 .. sizeX) {
+				additionTable[i] = HeapVector<MyClass> new(sizeY)
+				for (j in 0 .. sizeY) {
+					additionTable[i][j] = MyClass new(i + j)
+				}
```
Comment (davidhesselbom) [inline test/collections/VectorTest.ooc]: No need for these brackets.

File: test/collections/VectorTest.ooc [hunk 2]
```diff
@@ -214,12 +220,27 @@ VectorTest: class extends Fixture {
 		*/
 
 		this add("nested heap vector", func {
-			sizeX := 2
-			sizeY := 3
-			heapVector := HeapVector<HeapVector<MyCover>> new(sizeX)
-			for (i in 0 .. sizeX)
-				heapVector[i] = HeapVector<MyCover> new(sizeY)
-			heapVector free()
```
Comment (davidhesselbom) [inline test/collections/VectorTest.ooc]: Why was this removed? Keep it and put another test function.

File: test/collections/VectorTest.ooc [hunk 3]
```diff
@@ -214,12 +220,27 @@ VectorTest: class extends Fixture {
...
+			sizeX := 10
+			sizeY := 10
+			additionTable := HeapVector<HeapVector<MyClass>> new(sizeX)
```
Comment (simonmika) [inline test/collections/VectorTest.ooc]: Should be `HeapVector<Vector<MyClass>> new(sizeX)`.

File: test/collections/VectorTest.ooc [hunk 4]
```diff
@@ -214,12 +220,27 @@ VectorTest: class extends Fixture {
...
+			for (i in 0 .. sizeX) {
+				additionTable[i] = HeapVector<MyClass> new(sizeY)
+				for (j in 0 .. sizeY) {
+					additionTable[i][j] = MyClass new(i + j)
+				}
+			}
+			// Array access requires casting to the correct type when a collection holds a generic collection
```
Comment (simonmika) [inline test/collections/VectorTest.ooc]: Why?

File: test/collections/VectorTest.ooc [hunk 5]
```diff
@@ -214,12 +220,27 @@ VectorTest: class extends Fixture {
...
+			// 2 + 4 = 6
+			expect((additionTable[2] as HeapVector<MyClass>)[4] content, is equal to(6))
```
Comment (simonmika) [inline test/collections/VectorTest.ooc]: Why not `as Vector<MyClass>`?

File: test/collections/VectorTest.ooc (hunk 2 above)
Comment (davidpiuva) [inline test/collections/VectorTest.ooc]: It was the same code being extended and renamed.

File: test/collections/VectorTest.ooc (hunk 4 above)
Comment (davidpiuva) [inline test/collections/VectorTest.ooc]: Compilation will fail when type "T" is compared with "HeapVector<MyClass>".

I looked at the compiler's source code and found nothing to handle this case. Lots of comments about missing features and rare cases that are not handled however.

File: test/collections/VectorTest.ooc (hunk 2 above)
Comment (davidhesselbom) [inline test/collections/VectorTest.ooc]: The code you removed uses a `HeapVector<HeapVector<MyCover>>`, the new code uses a `HeapVector<HeapVector<MyClass>>`. Don't you think we should test both?

File: test/collections/VectorTest.ooc (hunk 5 above)
Comment (davidpiuva) [inline test/collections/VectorTest.ooc]: No reason. I will change it and see if it works.

File: test/collections/VectorTest.ooc (hunk 3 above)
Comment (davidhesselbom) [inline test/collections/VectorTest.ooc]: I didn't know that, thanks.

File: test/collections/VectorTest.ooc (hunk 4 above)
Comment (davidhesselbom) [inline test/collections/VectorTest.ooc]: I wouldn't expect it to get fixed by Amos anytime soon, so we'd probably have to fix it ourselves.

**Comment by davidpiuva** [comment]
test.sh passed all tests on my computer but why did Travis fail?

**Comment by davidhesselbom** [comment]
magic failed on Travis, not test.sh.

**Comment by davidpiuva** [comment]
Ok, I will run Magic again.

**Comment by davidhesselbom** [comment]
#377 fixes those errors, so ignore it for now.

**Comment by davidpiuva** [comment]
ok
